src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -0,0 +1,70 @@
+// <auto-generated/>
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    internal static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, Program.MyClass obj) => BindCore(configuration, ref obj);
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+        if (configuration.GetSection("MyString").Value is string stringValue0) { obj.MyString = stringValue0; }
+        if (configuration.GetSection("MyInt").Value is string stringValue1) { obj.MyInt = int.Parse(stringValue1); }
```

Comment:
A couple things here:

Parsing:
1. In order to be consistent with the reflection based behavior, we need to catch the int.Parse exception thrown here, and wrap it in a `InvalidOperationException` and add the configuration path of the invalid data. This allows devs to easily know where the invalid data exists from the exception message.
2. This should be `int.Parse(valueString, NumberStyles.Integer, NumberFormatInfo.InvariantInfo);`

See the following for how I did it when manually writing the bind code for Console Logging: https://github.com/dotnet/runtime/blob/967250cc86b4c613015066b47b882e7424fdb590/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerOptions.cs#L222-L239